### DISCUSSION TITLE
weird output from my script

### DISCUSSION THREAD

**Discussion by laurel-nave-powers** [sr320/course-fish546-2021#98] [Q&A] [answered]
Hello, I just ran a script and am not sure what the output means. It seems to have some errors but it also says that it was successful and that the deduplicated reads were written out to where I specified. However, when I go to the specified file it is empty with no output. I would appreciate some help wading through the error/success message that I get after running the script. 
Here is the notebook I'm working from with the long message at the end: https://github.com/fish546-2021/Laurel-genes/blob/main/code/03_rmdup.ipynb

**Comment by sr320** [comment]
A few lines in it states it cannot open

`/Users/laurelnave-powers/Desktop/GitHub/Laurel-genes/analyses/trimmed/HIPPLAT_NEF95054_S2_R1.fq_R1.fq`

I doubt this file actually exists? based on the name.

**Reply by laurel-nave-powers** [reply]
Yeah, I saw that too. I don't have any files with that name so I'm not sure why it is saying that. 
Why would it go on to say that it wrote out the deduplicated reads if it couldn't find the files?

**Reply by sr320** [reply]
That gets into how the perl script is written... https://github.com/fish546-2021/Laurel-genes/blob/main/scripts/rmdup.pl

**Comment by kubu4** [comment, marked as answer]
Adding this info for posterity. Per discussion in class, you had already figured out to eliminate the `_R1.fq` and `_R2.fq` from the end of your samples names list file. Just need to let it run longer and see what happens.

On that subject, another way to monitor progress/activity is to frequently run `ls -ltrh` in your output directory. This will sort things by time (newest on the bottom of the list) and you can look at the date/time listed to see if files have been updated very recently. You can also use the `top` command to look at what activity your computer is dealing with - presumably, you'd see a line with your `perl` command towards the top of that list.
